### PR TITLE
csv2md: init at 1.3.0

### PR DESCRIPTION
**Pull request by t4ccer**
## Description of changes

Added [csv2md](https://github.com/lzakharov/csv2md)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/csv2md/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ];
+
+  pythonImportsCheck = [ "csv2md" ];
+
+  meta = {
+    description = "Command line tool for converting CSV files into Markdown tables";
+    homepage = "https://github.com/lzakharov/csv2md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ t4ccer ];
+    mainProgram = "csv2md";
```
Comment (t4ccer) [inline pkgs/development/python-modules/csv2md/default.nix:29]: Both a library and CLI tool

File: pkgs/development/python-modules/csv2md/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "lzakharov";
+    repo = "csv2md";
+    rev = "v${version}";
+    hash = "sha256-INwZRbyJmyDd5SV7PBdqUc/3FO91upTgaxBfP/Giaw4=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (anthonyroussel) [inline pkgs/development/python-modules/csv2md/default.nix:18]:
```suggestion
  build-system = [
```

I think `setuptools` should be in `build-system` or `nativeBuildInputs` (now deprecated)

File: pkgs/development/python-modules/csv2md/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    setuptools
+  ];
+
+  pythonImportsCheck = [ "csv2md" ];
+
+  meta = {
+    description = "Command line tool for converting CSV files into Markdown tables";
```
Comment (anthonyroussel) [inline pkgs/development/python-modules/csv2md/default.nix:25]:
```suggestion
    changelog = "https://github.com/lzakharov/csv2md/releases/tag/v${version}";
    description = "Command line tool for converting CSV files into Markdown tables";
```

Maybe we could add meta.changelog here.
What do you think?

File: pkgs/development/python-modules/csv2md/default.nix (hunk 2 above)
Comment (t4ccer) [reply pkgs/development/python-modules/csv2md/default.nix:18]: How about `format = "setuptools";` and removing `propagatedBuildInputs` completely? Seems the cleanest but [manual](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md) doesn't mention it as possibility.

**Review by anthonyroussel — APPROVED** [review]
Builds and runs fine on x86_64-linux and aarch64-darwin :heavy_check_mark: 

If you want to make csv2md available as `csv2md` python application, you can add this in the `pkgs/top-level/all-packages.nix` as well :)

```nix
csv2md = with python3.pkgs; toPythonApplication csv2md;
```

**Comment by t4ccer** [comment]
Yeah, that's a good idea.

BTW, is there any docs whether we should prefer `python3Packages` or `python3.pkgs`?

**Comment by t4ccer** [comment]
I'll use https://github.com/NixOS/nixpkgs/issues/295031#issuecomment-1988792801 as my point of reference

**Review by anthonyroussel — APPROVED** [review]

**Pull request merged**